### PR TITLE
Update version of XEP-0045: Multi-User Chat

### PR DESCRIPTION
**Pull request by jacekwegr**
This PR updates implementation of MUC. 
The changes between version 1.25 and 1.34.5 include:
- Changes in the types of room config options
- Introduction of `<x/>` element in MUC-PMs
- Requirement to maintain the `id` attribute
- Removal of Group Chat 1.0 compatibility
- Use of a delay element in the initial subject message

Link to the full list of changes: https://xmpp.org/extensions/xep-0045.html#appendix-revs

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/179716) / small_tests / 1c5368a287e48b38d51abfcb2fe1e6e40e0ce7d3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4054/179716/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4054/179716/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-07-04_14.38.28/lib.mongooseim.logs/run.2023-07-04_14.38.29/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/179834) / small_tests / 743cc38dcf7bd4288597b05eada2bc193aac9cdf
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4054/179834/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4054/179834/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-07-05_13.59.35/lib.mongooseim.logs/run.2023-07-05_13.59.36/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/179985) / small_tests / 48121cde934533b65910fe93d85d4796fc46d932
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4054/179985/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4054/179985/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-07-07_14.14.44/lib.mongooseim.logs/run.2023-07-07_14.14.45/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/180708) / small_tests / 0bb0ec7c193cb5e3f52507a9c7200d1a9e340152
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4054/180708/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4054/180708/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-07-13_08.52.44/lib.mongooseim.logs/run.2023-07-13_08.52.45/suite.log.html)

File: big_tests/tests/muc_SUITE.erl [hunk 1]
```diff
@@ -4748,16 +4859,20 @@ print_next_message(User) ->
 print(Element) ->
     error_logger:info_msg("~n~p~n", [Element]).
 
-%Groupchat 1.0 protocol
+%Basic MUC protocol
 stanza_groupchat_enter_room(Room, Nick) ->
-    stanza_to_room(escalus_stanza:presence(<<"available">>), Room, Nick).
+    stanza_to_room(
+        escalus_stanza:presence(  <<"available">>,
```
Comment (JanuszJakubiec) [inline big_tests/tests/muc_SUITE.erl:4865]:
```suggestion
        escalus_stanza:presence(<<"available">>,
```

File: big_tests/tests/muc_SUITE.erl [hunk 2]
```diff
@@ -4748,16 +4859,20 @@ print_next_message(User) ->
...
+                                [#xmlel{ name = <<"x">>, attrs=[{<<"xmlns">>, <<"http://jabber.org/protocol/muc">>}]}]),
+        Room, Nick).
 
 
 stanza_groupchat_enter_room_no_nick(Room) ->
-    stanza_to_room(escalus_stanza:presence(<<"available">>), Room).
-
+    stanza_to_room(
+        escalus_stanza:presence(  <<"available">>,
```
Comment (JanuszJakubiec) [inline big_tests/tests/muc_SUITE.erl:4872]:
```suggestion
        escalus_stanza:presence(<<"available">>,
```

File: big_tests/tests/muc_SUITE.erl [hunk 3]
```diff
@@ -4748,16 +4859,20 @@ print_next_message(User) ->
 print(Element) ->
     error_logger:info_msg("~n~p~n", [Element]).
 
-%Groupchat 1.0 protocol
+%Basic MUC protocol
 stanza_groupchat_enter_room(Room, Nick) ->
-    stanza_to_room(escalus_stanza:presence(<<"available">>), Room, Nick).
+    stanza_to_room(
+        escalus_stanza:presence(  <<"available">>,
+                                [#xmlel{ name = <<"x">>, attrs=[{<<"xmlns">>, <<"http://jabber.org/protocol/muc">>}]}]),
```
Comment (JanuszJakubiec) [inline big_tests/tests/muc_SUITE.erl:4866]:
```suggestion
                                [#xmlel{name = <<"x">>, attrs=[{<<"xmlns">>, <<"http://jabber.org/protocol/muc">>}]}]),
```

File: big_tests/tests/muc_SUITE.erl [hunk 4]
```diff
@@ -4748,16 +4859,20 @@ print_next_message(User) ->
...
+        Room, Nick).
 
 
 stanza_groupchat_enter_room_no_nick(Room) ->
-    stanza_to_room(escalus_stanza:presence(<<"available">>), Room).
-
+    stanza_to_room(
+        escalus_stanza:presence(  <<"available">>,
+                                [#xmlel{ name = <<"x">>, attrs=[{<<"xmlns">>, <<"http://jabber.org/protocol/muc">>}]}]),
```
Comment (JanuszJakubiec) [inline big_tests/tests/muc_SUITE.erl:4873]:
```suggestion
                                [#xmlel{name = <<"x">>, attrs=[{<<"xmlns">>, <<"http://jabber.org/protocol/muc">>}]}]),
```

File: big_tests/tests/muc_SUITE.erl [hunk 5]
```diff
@@ -3131,8 +3201,9 @@ disco_info_locked_room(Config) ->
                      Alice, stanza_to_room(escalus_stanza:iq_get(?NS_DISCO_INFO,[]), RoomName)),
 
         %% THEN receives MUC features
-        Namespaces = [?NS_MUC, <<"muc_public">>, <<"muc_temporary">>, <<"muc_open">>,
-                     <<"muc_semianonymous">>, <<"muc_moderated">>, <<"muc_unsecured">>],
+        Namespaces = [?NS_MUC, ?NS_MUC_STABLE_ID,<<"muc_public">>, <<"muc_temporary">>,
```
Comment (JanuszJakubiec) [inline big_tests/tests/muc_SUITE.erl:3204]:
```suggestion
        Namespaces = [?NS_MUC, ?NS_MUC_STABLE_ID, <<"muc_public">>, <<"muc_temporary">>,
```

**Review by JanuszJakubiec — CHANGES_REQUESTED** [review]
Looks good. I've added minor comments

File: big_tests/tests/muc_SUITE.erl [hunk 6]
```diff
@@ -4795,6 +4910,14 @@ generate_room_addrs(FromN, ToN) ->
 stanza_message_to_room(Room, Payload) ->
     stanza_to_room(#xmlel{name = <<"message">>, children = Payload}, Room).
 
+stanza_private_muc_message(To, Msg) ->
+        #xmlel{name = <<"message">>,
+            attrs = [{<<"to">>, To}, {<<"type">>, <<"chat">>}],
+            children = [#xmlel{name = <<"body">>,
+                                children = [#xmlcdata{content = Msg}]},
```
Comment (JanuszJakubiec) [inline big_tests/tests/muc_SUITE.erl:4917]:
```suggestion
                               children = [#xmlcdata{content = Msg}]},
```

File: big_tests/tests/muc_SUITE.erl [hunk 7]
```diff
@@ -4795,6 +4910,14 @@ generate_room_addrs(FromN, ToN) ->
...
+                        #xmlel{name = <<"x">>,
+                                attrs = [{<<"xmlns">>, <<"http://jabber.org/protocol/muc#user">>}]}]}.
```
Comment (JanuszJakubiec) [inline big_tests/tests/muc_SUITE.erl:4919]:
```suggestion
                               attrs = [{<<"xmlns">>, <<"http://jabber.org/protocol/muc#user">>}]}]}.
```

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/180878) / small_tests / 08cd1fc21dd1a26b220700634b1b9318c89a56c8
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4054/180878/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4054/180878/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-07-14_12.36.48/lib.mongooseim.logs/run.2023-07-14_12.36.49/suite.log.html)

File: big_tests/tests/muc_SUITE.erl [hunk 8]
```diff
@@ -4748,21 +4859,25 @@ print_next_message(User) ->
...
 
-%Basic MUC protocol
 stanza_muc_enter_password_protected_room(Room, Nick, Password) ->
     stanza_to_room(
-        escalus_stanza:presence(  <<"available">>,
-                                [#xmlel{ name = <<"x">>, attrs=[{<<"xmlns">>, <<"http://jabber.org/protocol/muc">>}],
-                                             children=[#xmlel{name = <<"password">>, children = [#xmlcdata{content=[Password]}]} ]}]),
+        escalus_stanza:presence(<<"available">>,
+                                [#xmlel{name = <<"x">>, attrs=[{<<"xmlns">>, <<"http://jabber.org/protocol/muc">>}],
+                                            children=[#xmlel{name = <<"password">>, children = [#xmlcdata{content=[Password]}]} ]}]),
```
Comment (JanuszJakubiec) [inline big_tests/tests/muc_SUITE.erl:4880]:
```suggestion
                                            children = [#xmlel{name = <<"password">>, children = [#xmlcdata{content=[Password]}]} ]}]),
```

File: big_tests/tests/muc_SUITE.erl [hunk 9]
```diff
@@ -4748,21 +4859,25 @@ print_next_message(User) ->
 print(Element) ->
     error_logger:info_msg("~n~p~n", [Element]).
 
-%Groupchat 1.0 protocol
+%Basic MUC protocol
 stanza_groupchat_enter_room(Room, Nick) ->
-    stanza_to_room(escalus_stanza:presence(<<"available">>), Room, Nick).
+    stanza_to_room(
+        escalus_stanza:presence(<<"available">>,
+                                [#xmlel{name = <<"x">>, attrs=[{<<"xmlns">>, <<"http://jabber.org/protocol/muc">>}]}]),
```
Comment (JanuszJakubiec) [inline big_tests/tests/muc_SUITE.erl:4866]:
```suggestion
                                [#xmlel{name = <<"x">>, attrs = [{<<"xmlns">>, <<"http://jabber.org/protocol/muc">>}]}]),
```

**Review by JanuszJakubiec — CHANGES_REQUESTED** [review]
After rereading the file, I've found some more minor issues.

**Review by JanuszJakubiec — APPROVED** [review]
Looks good

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/180906) / small_tests / 0d9d69076ada7223a511029d884b496f4dae0e76
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4054/180906/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4054/180906/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-07-14_13.06.14/lib.mongooseim.logs/run.2023-07-14_13.06.15/suite.log.html)

**Pull request merged**